src/scripts/main.js

```diff
@@ -1,3 +1,31 @@
 'use strict';
 
-// write code here
+const employees = [...document.querySelectorAll('li')];
+
+function sortList(list) {
+  return list.sort((a, b) => (
+    b.dataset.salary.replace(/\D/g, '') - a.dataset.salary.replace(/\D/g, '')));
+}
+
+sortList(employees);
+
+function getEmployees(list) {
+  const result = [];
+
+  list.forEach(element => {
+    result.push({
+      name: element.innerText,
+      position: element.dataset.position,
+      salary: element.dataset.salary,
+      age: element.dataset.age,
+    });
+  });
+
+  return result;
```

Comment:
```suggestion
  const result = [];

  list.forEach(element => {
    result.push({
      name: element.innerText,
      position: element.dataset.position,
      salary: element.dataset.salary,
      age: element.dataset.age,
    });
  });

  return result;
```

You can get rid of variable result, just make copy of your list, which you take as parametr, and return the result of iteration through it with map function